staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -1977,6 +1977,10 @@ type Probe struct {
 	// Defaults to 3. Minimum value is 1.
 	// +optional
 	FailureThreshold int32 `json:"failureThreshold,omitempty" protobuf:"varint,6,opt,name=failureThreshold"`
+	// Maximum number of failures allowed for the probe to be considered failed before having succeeded once.
```

Comment:
We need to make it clearer that this is only while waiting for the pod to start up. We need to document the behavior when kubelet restarts (at a minimum); I was thinking it was OK to restart the count in that case but with fresh eyes it's not sounding like a good idea to me any more. :/

I'll suggest some wording in a bit.

We also need to document and/or enforce compatibility/version skew guarantees.